### PR TITLE
Provision scsi disks for libvirt domains

### PR DESCRIPTION
**Pull request by trewest**
Changes:

- Changes terraform provisioned disks to scsi
- Change libvirt_domain machine type to q35
- Renames `consolemodel.xsl` to `console_cdrom_model.xsl`
- Adds xsl to update cdrom bus to sata rather than ide

### CODE REVIEW

File: terraform_files/baremetal/console_cdrom_model.xsl [hunk 1]
```diff
@@ -15,4 +15,10 @@
         <log file="/var/log/libvirt/qemu/{/domain/name}-console.log" append="on"/>
       </xsl:copy>
   </xsl:template>
+
+  <xsl:template match="/domain/devices/disk[@device='cdrom']/target/@bus">
+  <xsl:attribute name="bus">
```
Comment (osherdp) [inline terraform_files/baremetal/console_cdrom_model.xsl:20]: Any chance you can fix indentation?
Thanks!

File: terraform_files/baremetal_host/main.tf [hunk 2]
```diff
@@ -68,8 +69,10 @@ resource "libvirt_domain" "host" {
     }
   }
 
+  machine = "q35"
+
   xml {
-    xslt = file("consolemodel.xsl")
+    xslt = file("console_cdrom_model.xsl")
```
Comment (osherdp) [inline terraform_files/baremetal_host/main.tf:75]:
```suggestion
    xslt = file("../baremetal_host/console_cdrom_model.xsl")
```
then those other files under ``terraform_files/baremetal``, ``terraform_files/none`` & ``terraform_files/baremetal_infra_env`` can safely be deleted and we can save ourselves the need to update it everywhere

File: terraform_files/baremetal_host/main.tf [hunk 3]
```diff
@@ -68,8 +69,10 @@ resource "libvirt_domain" "host" {
     }
   }
 
+  machine = "q35"
```
Comment (osherdp) [inline terraform_files/baremetal_host/main.tf:72]: I think we should leave default behavior as-is and just allow templating this stuff if requested
You can never know what other environments are supporting and what's not, so even deciding to transition should be done gradually

**Review by osherdp — CHANGES_REQUESTED** [review]

File: terraform_files/baremetal_host/main.tf (hunk 2 above)
Comment (trewest) [reply terraform_files/baremetal_host/main.tf]: done

File: terraform_files/baremetal_host/main.tf (hunk 3 above)
Comment (osherdp) [reply terraform_files/baremetal_host/main.tf:72]: the same way we're templating ``vcpu``, ``memory`` and ``name`` in this file (to name a few). Those values are being set based on user input or other calculations

Comment (osherdp) [reply terraform_files/baremetal_host/main.tf:72]: For more info, please read: https://github.com/openshift/assisted-test-infra/blob/master/GUIDE.md#adding-a-new-e2e-flow

Comment (trewest) [reply terraform_files/baremetal_host/main.tf:72]: okay, I see what you are saying now

**Comment by trewest** [comment]
Relies on some commits provided by: https://github.com/openshift/assisted-test-infra/pull/1709

**Pull request closed without merging**